src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest.kt

```diff
@@ -910,59 +906,440 @@ class OffenderResourceIntTest : ResourceTest() {
+        .headers(setAuthorisation("WAI_USER", listOf("ROLE_ADD_CASE_NOTES")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(
+          """
+            {
+              "type": "GEN",
+              "subType": "OSE",
+              "text": "A new case note",
+              "occurrenceDateTime": "2017-04-14T10:15:30"      
+            }
+          """,
+        )
+        .exchange()
+        .expectStatus().isOk
+        .expectBody(CaseNote::class.java).returnResult().responseBody
+
+      removeCaseNoteCreated(response.caseNoteId)
+    }
+
+    @Test
+    fun `returns 200 when has override role ROLE_ADD_CASE_NOTES`() {
+      val response = webTestClient.post().uri("/api/offenders/$OFFENDER_NUMBER/case-notes")
+        .headers(setAuthorisation(listOf("ROLE_ADD_CASE_NOTES")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(caseNote)
+        .exchange()
+        .expectStatus().isOk
+        .expectBody(CaseNote::class.java).returnResult().responseBody
+
+      removeCaseNoteCreated(response.caseNoteId)
+    }
+
+    @Test
+    fun testCreateMovedCellCaseNote() {
+      val caseNote = webTestClient.post().uri("/api/offenders/A9876RS/case-notes")
+        .headers(setAuthorisation(listOf()))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(
+          """
+            {
+              "type": "MOVED_CELL",
+              "subType": "BEH",
+              "text": "This is a test comment"
+            }
+          """,
+        )
+        .exchange()
+        .expectStatus().isOk
+        .expectBody(CaseNote::class.java).returnResult().responseBody
+
+      removeCaseNoteCreated(caseNote.caseNoteId)
+    }
+
+    @Test
+    fun `A case note is successfully created for an offender`() {
+      val caseNote = webTestClient.post().uri("/api/offenders/A1176RS/case-notes")
+        .headers(setAuthorisation(listOf()))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(
+          """
+            {
+              "type": "OBSERVE",
+              "subType": "OBS_GEN",
+              "text": "A new case note",
+              "occurrenceDateTime": "2017-04-14T10:15:30"      
+            }
+          """,
+        )
+        .exchange()
+        .expectStatus().isOk
+        .expectBody(CaseNote::class.java).returnResult().responseBody
+
+      assertThat(caseNote.caseNoteId).isGreaterThan(0)
+      assertThat(caseNote.source).isEqualTo(caseNoteSource)
+      assertThat(caseNote.type).isEqualTo("OBSERVE")
+      assertThat(caseNote.subType).isEqualTo("OBS_GEN")
+      assertThat(caseNote.text).isEqualTo("A new case note")
+      assertThat(caseNote.occurrenceDateTime).isEqualTo("2017-04-14T10:15:30")
+      assertThat(caseNote.creationDateTime).isNotNull()
+
+      removeCaseNoteCreated(caseNote.caseNoteId)
+    }
+
+    @Test
+    fun `A second case note is successfully created for booking`() {
```

Comment:
Not sure how this test is different to the previous test?

Reply:
ha ha it's not - it was just pulled in from the previous tests - shall I just remove it?

Reply:
after conversation with Mike - will change the test in a separate PR to add multiple case notes